### ISSUE TITLE
Option for the `confirm` option to always confirm Neovim exit, regardless of buffer status

### ISSUE THREAD

**Issue by mrjones2014** [neovim/neovim#17256]
### Feature already in Vim?

No

### Feature description

An option like `:set confirm` that will _always_ confirm that you want to exit Neovim, regardless of whether you have unsaved buffers or not. I've tried implementing something like this as a plugin but it doesn't really work.

**Comment by mrjones2014** [comment]
To add some more context: the reason I want something like this is the following:

I work in a huge Rust monorepo. In such a huge project, `rust-analyzer` takes like 1-2 full minutes to initialize. For that reason, I don't want to kill Neovim to get back to the terminal, instead I want to leave Neovim open and either use a terminal buffer or a new tmux window to get to my terminal, so that I don't have to close and re-open Neovim, then wait 1-2 minutes for `rust-analyzer` to re-initialize.

Does that make sense? I'm open to alternative solutions, but this seems like a simple solution to me.

**Comment by vimpostor** [comment]
> An option like `:set confirm` that will _always_ confirm that you want to exit Neovim, regardless of whether you have unsaved buffers or not.

Just use `:qa!`.

As for your usecase about having to quit vim to get to the terminal, you can either start a terminal in neovim (`:term`), you can suspend neovim temporarily (`C-Z`), or you can use an external terminal multiplexer like `tmux`.

**Comment by zeertzjq** [comment]
OP meant always confirm, not never confirm.

**Comment by mrjones2014** [comment]
>As for your usecase about having to quit vim to get to the terminal, you can either start a terminal in neovim (:term), you can suspend neovim temporarily (C-Z), or you can use an external terminal multiplexer like tmux.

Yeah, I'm using `tmux` for this primarily. But sometimes by mistake, just out of muscle memory, I might type `:q` or `:qa` -- in that case I'd like Neovim to confirm before exiting.

**Comment by vimpostor** [comment]
Maybe an option could be added to the VimLeavePre command that can
prevent vim from quitting.

However you can also "solve" your problem with the following hacky
workaround:

:autocmd CmdlineLeave * set modified

> Yeah, I'm using `tmux` for this primarily. But sometimes by mistake, just out of muscle memory, I might type `:q` or `:qa` -- in that case I'd like Neovim to confirm before exiting.

**Comment by mrjones2014** [comment]
`:autocmd ExitPre * set modified` kinda works, but it's a bit weird because really it's asking me if I want to save the file before exit. So the context is a bit off. But I guess it works in the mean time.

**Comment by alexmozaidze** [comment]
There's a [plugin](https://github.com/yutkat/confirm-quit.nvim) that does this.
It works with simple `:q`, but be wary that it doesn't support keymaps (yet), see [#1](https://github.com/yutkat/confirm-quit.nvim/issues/1) for that.

**Comment by heygarrett** [comment]
I'd also love to have this as an option. In my case, I'd like to prevent myself from saving and closing a commit message buffer too quickly if I haven't noticed diagnostics from commitlint.
